### ISSUE TITLE
exponent operator returning type Any

### ISSUE THREAD

**Issue by bschnurr** [microsoft/pyright#4206]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**

**To Reproduce**
Steps to reproduce the behavior.

```python
tmp = int(9) ** 0.5  # tmp: Any
tmp = 9 ** 0.5 # tmp: float
```

**Expected behavior**

expected both cases to be type 'float'

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
![image](https://user-images.githubusercontent.com/1946977/202783662-999bda94-3a37-4443-a5dd-5da6741f392c.png)

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

**Additional context**

**Comment by bschnurr** [comment]
```python
tmp = int(9) ** 1      # tmp: float
```

**Comment by bschnurr** [comment]
I'll try to fix it there. thanks

**Comment by erictraut** [comment]
I don't think there's a good fix. This has been discussed at length among the typeshed maintaners, and `Any` is the best answer they can provide in this case. Search for closed typeshed issues for more details.

**Comment by bschnurr** [comment]
strange

```python
    @overload
    def __pow__(self, __x: float, __modulo: None = ...) -> float: ...
```

this solves it
and works for both

```python

tmp = int(9) ** 0.5

print(tmp)  # prints 3.0

tmp = int(9) ** 1.0

print(tmp)   #prints 9.0

```

**Comment by bschnurr** [comment]
I'm trying to fix a datascience code example which fails to retain the 'Tensor' type

`W1 = torch.randn((inputDemension, hiddenLayerSize), generator=g) * (5.0/3.0) / (inputDemension * hiddenLayerSize ** 0.5)`

**Comment by bschnurr** [comment]
"As a historical note, in 2022 pow() is now typed with fifteen alternative type signatures" lol

**Comment by bschnurr** [comment]
thinking codeAction on '**' expressions returning 'Any' to insert an 'int' or 'float' cast

**Comment by erictraut** [comment]
You're proposing to do this for any expression that includes a `**` operator? Or only if the LHS includes an `int` type? From a type checking perspective, `Any` is a reasonable answer. What are you trying to achieve by adding a cast? Is the concern mainly about completion suggestions not working for the `Any` result?

**Comment by bschnurr** [comment]
ya. just trying to retain completions on Tensors

**Comment by bschnurr** [comment]
and typing system doesn't support 'Literal[0.5]' for stub overloads
